### PR TITLE
Added: Furnaces

### LINKED ISSUE #119: Block placement (stairs, etc.)
BlockTypes representing stairs, slabs, logs, leaves, ladders, and other assorted blocks with facing need to be created to allow these to be placed correctly. The `placeBlock` function should be overridden to modify the passed BlockState to set data based on the player's facing or the face placed against as needed.

Blocks involved:
- [x] Torches
- [ ] Redstone torches
- [x] Logs
- [ ] Leaves
- [x] Slabs
- [ ] Tall flowers
- [ ] Pistons
- [x] Stairs
- [X] Sign posts and wall signs
- [x] Doors
- [ ] Rails
- [x] Ladders
- [x] Furnaces
- [x] Chests / Ender chests
- [x] Dispensers
- [x] Droppers
- [x] Hoppers
- [x] Levers
- [x] Buttons
- [ ] Pumpkins / Jack o'Lanterns
- [ ] Redstone repeaters / comparators
- [x] Trap doors
- [x] Vines
- [x] Fence gates
- [ ] Cocoa beans
- [ ] Tripwire hooks
- [x] Heads
- [x] Hay
- [x] Quartz
- [x] Beds
- [ ] Anvils
- [x] Banners

Edit (turt2live): Issue scope expanded to general placement problems including data not being set (leaves, etc) and not just facing directions.

### PR DESCRIPTION
**Pull request by anna-is-cute**
Partially fixes #119. No functionality besides placement.

Requires #176.

### CODE REVIEW

**Comment by TomCaserta** [comment]
Looks good. Probably be worthwhile to move the getOppositeBlockFace to the BlockType class since other blocks (such as Ender Portal Frame) could use this for its direction.

File: src/main/java/net/glowstone/block/entity/TEDispenser.java [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package net.glowstone.block.entity;
+
+import net.glowstone.block.GlowBlock;
+import org.bukkit.event.inventory.InventoryType;
+
+public class TEDispenser extends TEContainer {
+
+    public TEDispenser(GlowBlock block) {
+        super(block, InventoryType.DISPENSER);
+    }
```
Comment (turt2live) [inline src/main/java/net/glowstone/block/entity/TEDispenser.java:10]: Your other files have a blank line after this - is this intentional?

File: src/main/java/net/glowstone/block/entity/TEFurnace.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+import net.glowstone.block.GlowBlock;
+import org.bukkit.event.inventory.InventoryType;
+
+public class TEFurnace extends TEContainer {
+
+    public TEFurnace(GlowBlock block) {
+        super(block, InventoryType.FURNACE);
+        setSaveId("Furnace");
+    }
```
Comment (turt2live) [inline src/main/java/net/glowstone/block/entity/TEFurnace.java:11]: Your other files have a blank line after this - is this intentional?

Comment (anna-is-cute) [inline src/main/java/net/glowstone/block/entity/TEFurnace.java:11]: Does it matter?

File: src/main/java/net/glowstone/block/blocktype/BlockContainer.java [hunk 3]
```diff
@@ -23,4 +24,41 @@ public boolean blockInteract(GlowPlayer player, GlowBlock block, BlockFace face,
...
 
+    /**
+     * Gets the BlockFace opposite of the direction the location is facing. Usually used to set the way container blocks
+     * face when being placed.
+     *
+     * @param location Location to get opposite of
+     * @param inverted If up/down should be used
+     * @return Opposite BlockFace or EAST if pitch is invalid
+     */
+    public BlockFace getOppositeBlockFace(Location location, boolean inverted) {
```
Comment (turt2live) [inline src/main/java/net/glowstone/block/blocktype/BlockContainer.java:35]: https://github.com/SpaceManiac/Glowstone/pull/166/files#r17219136

**Comment by SpaceManiac** [comment]
Merged in 8d9cf2f.

**Pull request closed without merging**